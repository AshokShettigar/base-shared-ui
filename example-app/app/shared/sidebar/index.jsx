import React from 'react';
import { Button } from 'platform-common-ui';
import BaseComponentView from 'app/components/base-component-view';

import { Sidebar } from '../index';

export default class SidebarExample extends BaseComponentView {
    state = {
        isOpen: false
    };

    handleOpen = () => {
        this.setState({ isOpen: true });
    };

    handleClose = () => {
        this.setState({ isOpen: false });
    };

    getOptions() {
        return (
            <div>
                <Button onClick={this.handleOpen}>Open sidebar</Button>
            </div>
        );
    }

    renderComponent() {
        return (
            <Sidebar
                position="right"
                title="Sidebar title"
                isOpen={this.state.isOpen}
                onDismiss={this.handleClose}
                headerClassName="header-class-name"
            >
                Sidebar content
            </Sidebar>
        );
    }
}
