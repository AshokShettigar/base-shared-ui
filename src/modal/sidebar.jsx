import PropTypes from 'prop-types';

const Sidebar = props => <div className="cn-modal-sidebar">{props.children}</div>;

Sidebar.displayName = 'ModalSidebar';

Sidebar.propTypes = {
    children: PropTypes.node.isRequired
};

export default Sidebar;
