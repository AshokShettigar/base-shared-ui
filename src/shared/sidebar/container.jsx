import React, { Component } from 'react';
import { string, bool, node, func, oneOf } from 'prop-types';
import ReactModal from 'react-modal';
import classNames from 'classnames';
import { isEmpty } from 'lodash';

import {
    SidebarHeaderSection,
    SidebarBodySection
} from './sections';
import styles from './container.scss';

class Sidebar extends Component {
    static propTypes = {
        title: string,
        isOpen: bool,
        shouldCloseOnOverlayClick: bool,
        onAfterOpen: func,
        onDismiss: func,
        onNavigateToNext: func,
        onNavigateToPrevious: func,
        className: string,
        headerClassName: string,
        overlayClassName: string,
        contentClassName: string,
        bodyOpenClassName: string,
        children: node,
        parentRef: node,
        position: oneOf(['left', 'right']),

        sidebarBodyRef: func
    };

    static defaultProps = {
        title: undefined,
        isOpen: false,
        shouldCloseOnOverlayClick: false,
        onAfterOpen: undefined,
        onDismiss: undefined,
        onNavigateToNext: undefined,
        onNavigateToPrevious: undefined,
        className: undefined,
        headerClassName: undefined,
        overlayClassName: undefined,
        contentClassName: undefined,
        bodyOpenClassName: 'sidebar-open',
        children: undefined,
        parentRef: undefined,
        position: 'right',
        sidebarBodyRef: undefined
    };

    state = {
        isOpen: this.props.isOpen
    };

    componentWillReceiveProps(nextProps) {
        if (nextProps.isOpen !== this.state.isOpen) {
            if (nextProps.isOpen) {
                this.handleOpen();
            } else {
                this.handleClose();
            }
        }
    }

    handleClose = () => {
        this.setState({ isOpen: false });

        if (this.props.onDismiss) {
            this.props.onDismiss();
        }
    };

    handleOpen() {
        this.setState({ isOpen: true });
    }

    parentSelector = () => this.props.parentRef;

    render() {
        const toggleClass = this.state.isOpen
            ? styles[`show-${this.props.position}`]
            : styles[`hide-${this.props.position}`];
        const overlayClass = isEmpty(this.props.parentRef)
            ? styles[`overlay-${this.props.position}`]
            : styles[`overlay-parent-${this.props.position}`];

        return (
            <ReactModal
                isOpen={this.state.isOpen}
                onAfterOpen={this.props.onAfterOpen}
                bodyOpenClassName={this.props.bodyOpenClassName}
                contentLabel={this.props.title}
                onRequestClose={this.handleClose}
                parentSelector={this.props.parentRef && this.parentSelector}
                portalClassName={classNames(styles.root, this.props.className)}
                overlayClassName={
                    classNames(
                        overlayClass,
                        styles[`sidebar-position-${this.props.position}`],
                        this.props.overlayClassName)
                }
                className={classNames(styles.content, this.props.contentClassName, toggleClass)}
                role="dialog"
                ariaHideApp={false}
                closeTimeoutMS={300}
                shouldCloseOnEsc
                shouldReturnFocusAfterClose
                shouldCloseOnOverlayClick={this.props.shouldCloseOnOverlayClick}
            >
                <SidebarHeaderSection
                    className={this.props.headerClassName}
                    onDismiss={this.handleClose}
                    onNavigateToNext={this.props.onNavigateToNext}
                    onNavigateToPrevious={this.props.onNavigateToPrevious}
                >
                    {this.props.title}
                </SidebarHeaderSection>
                <SidebarBodySection sidebarBodyRef={this.props.sidebarBodyRef}>
                    {this.props.children}
                </SidebarBodySection>
            </ReactModal>
        );
    }
}

export default Sidebar;
