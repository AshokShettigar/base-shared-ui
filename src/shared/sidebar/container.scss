@import '../scss/variables.scss';

.root {
  font-family: $ff-open-sans;
  font-size: $fs-tiny;
  color: $c-primary;
}

%overlay {
  display: flex;
  justify-content: center;
  position: fixed;
  z-index: $z-modal-window;
}

.overlay-right {
  @extend %overlay;

  top: 0;
  right: 0;
  bottom: 0;
}

.overlay-left {
  @extend %overlay;

  top: 0;
  left: 0;
  bottom: 0;
  width: 100%;
}

.overlay-parent-right {
  @extend %overlay;

  right: 0;
  height: 100%;
}

.overlay-parent-left {
  @extend %overlay;

  left: 0;
  height: 100%;
}

.content {
  max-width: 650px;
  width: 650px;
  height: 100%;
  background: white;
  display: flex;
  flex-direction: column;
  box-shadow: -2px 2px 1px 0 rgba(221, 221, 221, 0.5);

  &:focus {
    outline: none;
  }
}

.show-right {
  animation: animate-show-right .3s ease;
}

.hide-right {
  animation: animate-hide-right .3s ease;
}

.show-left {
  animation: animate-show-left .3s ease;
}

.hide-left {
  animation: animate-hide-left .3s ease;
}

@keyframes animate-show-right {
  from { transform: translateX(100%); }
  to { transform: translateX(0); }
}

@keyframes animate-hide-right {
   from { transform: translateX(0); }
   to { transform: translateX(100%); }
 }

@keyframes animate-show-left {
  from { transform: translateX(-100%); }
  to { transform: translateX(0); }
}

@keyframes animate-hide-left {
  from { transform: translateX(0); }
  to { transform: translateX(-100%); }
}

.sidebar-position-right {
  justify-content: flex-end;
}

.sidebar-position-left {
  justify-content: flex-start;
}
