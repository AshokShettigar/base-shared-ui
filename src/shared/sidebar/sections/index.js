export { default as SidebarHeaderSection } from './header';
export { default as SidebarBodySection } from './body';
