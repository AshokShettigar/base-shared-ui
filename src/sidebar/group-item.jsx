import PropTypes from 'prop-types';
import classNames from 'classnames';

// eslint-disable-next-line import/prefer-default-export
export const Item = function(props) {
    const className = classNames('cn-sidebar-group__item', props.className);
    return (
        <li className={className}>
            {props.children}
        </li>
    );
};

Item.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string
};
