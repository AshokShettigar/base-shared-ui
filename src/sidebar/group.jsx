import PropTypes from 'prop-types';
import classNames from 'classnames';

// eslint-disable-next-line import/prefer-default-export
export const Group = props => {
    const className = classNames('cn-sidebar-group', props.className);
    const titleClassName = classNames('cn-sidebar-group__title', props.titleClassName);
    /* eslint-disable jsx-a11y/interactive-supports-focus */
    return (
        <div className={className}>
            <span
                className={titleClassName}
                onClick={props.onChangeExpandCollapse}
                role="button"
            >
                {props.title}
            </span>
            {!props.isCollapsed && <ul className="cn-sidebar-group__list">{props.children}</ul>}
        </div>
    );
    /* eslint-disable jsx-a11y/interactive-supports-focus */
};

Group.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    isCollapsed: PropTypes.bool.isRequired,
    onChangeExpandCollapse: PropTypes.func.isRequired,
    title: PropTypes.node.isRequired,
    titleClassName: PropTypes.string
};

Group.defaultProps = {
    isCollapsed: false
};
