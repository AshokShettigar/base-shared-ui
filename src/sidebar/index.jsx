import PropTypes from 'prop-types';
import { Icon } from 'platform-common-ui';
import classNames from 'classnames';

// eslint-disable-next-line import/prefer-default-export
export const Sidebar = props => {
    const classes = classNames({
        'cn-sidebar': true,
        'cn-sidebar--closed': !props.isOpen
    });

    return (
        <div className={classes}>
            <div className="cn-sidebar__controls">
                <Icon
                    className="cn-sidebar__close-btn"
                    icon={props.isOpen ? 'close' : 'carat-right'}
                    onClick={props.onToggleSidebar}
                />
            </div>{ props.isOpen && <div className="cn-sidebar__content">{props.children}</div>}
        </div>
    );
};

Sidebar.propTypes = {
    children: PropTypes.node.isRequired,
    isOpen: PropTypes.bool.isRequired,
    onToggleSidebar: PropTypes.func.isRequired
};

Sidebar.displayName = 'Sidebar';

Sidebar.defaultProps = {
    isOpen: true
};
