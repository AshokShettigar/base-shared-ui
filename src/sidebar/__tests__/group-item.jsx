import { shallow } from 'enzyme';
import { Item } from 'platform-common-ui/sidebar/group-item';

describe('sidebar/group-item', function() {
    it('should render the group item component', function() {
        const wrapper = shallow(<Item>Item 1</Item>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the group item with custom item class name', function() {
        const wrapper = shallow(<Item className="my-class">Item 1</Item>);
        expect(wrapper).toMatchSnapshot();
    });
});
