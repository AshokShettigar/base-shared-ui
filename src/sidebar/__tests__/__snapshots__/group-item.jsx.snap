// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar/group-item should render the group item component 1`] = `
<li
  className="cn-sidebar-group__item"
  data-component="Item"
>
  Item 1
</li>
`;

exports[`sidebar/group-item should render the group item with custom item class name 1`] = `
<li
  className="cn-sidebar-group__item my-class"
  data-component="Item"
>
  Item 1
</li>
`;
